app/controllers/api/v1/admin/statuses_controller.rb

```diff
@@ -24,7 +24,14 @@ def destroy
       @status.discard_with_reblogs
       log_action :destroy, @status
       Tombstone.find_or_create_by(uri: @status.uri, account: @status.account, by_moderator: true)
+
+      if @status.with_media?
+        # Immediately remove public copy of media instead of waiting for
+        # the vacuum_orphaned_records job to take care of it later on
+        destroy_associated_media
```

Comment:
This same logic needs to also happen in `status_actions_controller.rb` when `type` is `delete`. See the [original API design doc](https://docs.google.com/document/d/11p55NqeffaonPrH87cYxpN5iv6ubUK-6C1hzRRVRRXU/edit#heading=h.oha0k6716u0w) for why the delete action is duplicated in two places.